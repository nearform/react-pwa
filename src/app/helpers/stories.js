const qs = require('qs')
const fetch = require('cross-fetch')

const isBrowser = typeof window !== 'undefined'
const HOST = isBrowser
  ? ''
  : process.env.NOW_URL || 'http://localhost:3000'

async function fetchStories ({ sort, filter, page } = {}) {
  const params = qs.stringify({
    sort,
    filter,
    page
  })
  const res = await fetch(`${HOST}/api/stories?${params}`)
  const data = await res.json()
  if (data.message && data.stack && data.statusCode) {
    throw data
  }
  return {data}
}

module.exports = {
  fetchStories
}
