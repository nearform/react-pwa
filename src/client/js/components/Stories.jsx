import { rem, quote } from 'csx'
import React from 'react'
import { style, keyframes } from 'typestyle'
import { debugClassName } from '../styles/common'

const calculateStartingNumber = pathname => {
  let currentPage = parseInt(pathname.split('page/')[1])
  if (!Number.isInteger(currentPage)) return 1
  return (currentPage - 1) * 30 + 1
}

const storiesListClassName = style(debugClassName('stories-list'), {
  display: 'grid',
  gridTemplateColumns: '100%',
  listStyleType: 'none',
  margin: 0,
  padding: 0,
})

const loadingAnimationName = keyframes({
  '100%': { transform: 'translateX(100%)' }
})

const storiesListItemClassName = style(debugClassName('stories-list-item'), {
  padding: '1em 1em 1em 0',
  display: 'grid',
  gridRowGap: '.5em',
  gridTemplateColumns: '53px',
  backgroundImage: 'linear-gradient(0deg, #fff 97%, #e2e2e2 100%)',
})


const storiesListItemClassNamePlaceholder = style(debugClassName('stories-list-item'), {
  padding: '1em 1em 1em 0',
  display: 'grid',
  gridRowGap: '.5em',
  gridTemplateColumns: '53px',
  backgroundImage: 'linear-gradient(0deg, #fff 97%, #e2e2e2 100%)',
  $nest: {
    '&::after': {
      content: "''",
      position: 'absolute',
      width: '100%',
      height: '100%',
      background: 'linear-gradient(90deg, rgba(0,0,0,0), rgba(255,255,255,0.15), rgba(0,0,0,0))',
      transform: 'translateX(-100%)',
      animationName: loadingAnimationName,
      animationDuration: '1.5s',
      animationIterationCount: 'infinite',
    }
  }
})

const storiesListIndexClassName = style(debugClassName('stories-list-index'), {
  gridColumnStart: '1',
  gridColumnEnd: '2',
  gridRowStart: '1',
  gridRowEnd: '3',
  background: '#ec1c2b',
  width: 20,
  height: 20,
  textAlign: 'center',
  padding: '1em',
  color: 'white',
})

const storiesListTitle = style(debugClassName('stories-list-title'), {
  padding: '0 1em 1em 1em',
  gridColumnStart: '2',
  gridColumnEnd: '3',
  gridRowStart: '1',
  gridRowEnd: '2',
})

const storiesListTitlePlaceholder = style(debugClassName('stories-list-title'), {
  padding: '0 1em 1em 1em',
  gridColumnStart: '2',
  gridColumnEnd: '3',
  gridRowStart: '1',
  gridRowEnd: '2',
  width: '66%'
})

const storiesListByLineClassName = style(debugClassName('stories-list-by-line'), {
  padding: '1em',
  gridColumnStart: '2',
  gridColumnEnd: '3',
  gridRowStart: '2',
  gridRowEnd: '3',
})

const placeholder = style({
  padding: 0,
  margin: 0,
  color: '#d4d4d4',
  background: '#d4d4d4',
})

const storiesListIndexClassNamePlaceholder = style(debugClassName('stories-list-index'), {
  gridColumnStart: '1',
  gridColumnEnd: '2',
  gridRowStart: '1',
  gridRowEnd: '3',
  background: '#d4d4d4',
  width: 20,
  height: 20,
  textAlign: 'center',
  padding: '1em',
  color: '#d4d4d4',
})

export function Stories({ data: stories, location }) {
  if (stories) {
    return (
      <ol className={storiesListClassName} start={calculateStartingNumber(location.pathname)}>
        {stories.filter(Boolean).map((story, index) => {
          const count = index === 0 ? calculateStartingNumber(location.pathname) : index + calculateStartingNumber(location.pathname);
          return (
          <li className={storiesListItemClassName} key={story.id}>
            <div className={storiesListIndexClassName}>{count}</div>
            <div className={storiesListTitle}><a href={story.url}>{story.title}</a></div>
            <div className={storiesListByLineClassName} suppressHydrationWarning>
              1 hour ago by {story.by.id} ({story.score} points)
            </div>
          </li>
        )
        })}
      </ol>
    )
  }

    return (<ol className={storiesListClassName}>
      {Array(5).fill({}).map((_, index) => (
        <li className={storiesListItemClassNamePlaceholder} key={index}>
          <div className={storiesListIndexClassNamePlaceholder}>{index + 1}</div>
          <div className={storiesListTitlePlaceholder}><p className={placeholder}>This is title</p></div>
          <div className={storiesListByLineClassName} suppressHydrationWarning>
            <p className={placeholder}>1 hour ago by dave (6 points)</p>
          </div>
        </li>
      ))}
    </ol>
  )
}
