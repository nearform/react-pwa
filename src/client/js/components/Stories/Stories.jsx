import React from 'react'
import { media, classes, stylesheet } from 'typestyle'
import { colors, loadingAnimation, placeholder, ergonomics } from '../../styles/common'

const calculateStartingNumber = pathname => {
  let currentPage = parseInt(pathname.split('page/')[1])
  if (!Number.isInteger(currentPage)) return 1
  return (currentPage - 1) * 30 + 1
}

const styles = stylesheet({
  storiesList: {
    display: 'grid',
    gridTemplateColumns: '100%',
    ...media(
      {
        minWidth: ergonomics.LAP.BEGINNING,
        maxWidth: ergonomics.LAP.END,
      },
      {
        gridTemplateColumns: 'repeat(2, 50%)',
      }
    ).$nest,
    ...media(
      {
        minWidth: ergonomics.DESK.BEGINNING,
      },
      {
        gridTemplateColumns: 'repeat(3, 33.333%)',
      }
    ).$nest,
  },
  storiesListItem: {
    position: 'relative',
    overflow: 'hidden',
    padding: '1em 0 0 0',
    display: 'grid',
    gridRowGap: '.5em',
    gridTemplateColumns: '53px',
    backgroundImage: `linear-gradient(0deg, white 97%, ${colors.LIGHT_GRAY} 100%)`,
    ...media(
      {
        minWidth: ergonomics.LAP.BEGINNING,
      },
      {
        borderRight: `1px solid ${colors.LIGHT_GRAY}`,
      }
    ),
  },
  storiesListIndex: {
    gridColumnStart: '1',
    gridColumnEnd: '2',
    gridRowStart: '1',
    gridRowEnd: '3',
    background: colors.NEARFORM_BRAND_MAIN,
    height: 40,
    textAlign: 'center',
    paddingTop: '1em',
    color: 'white',
  },
  storiesListTitle: {
    margin: 0,
    fontWeight: 'normal',
    fontSize: 'inherit',
    padding: '0 1em 1em 1em',
    gridColumnStart: '2',
    gridColumnEnd: '3',
    gridRowStart: '1',
    height: '4em',
    gridRowEnd: '2',
    $nest: {
      a: {
        color: 'black',
      },
    },
  },
  storiesListTitlePlaceholder: {
    width: '66%',
  },
  storiesListByLine: {
    padding: '.5em .5em .5em 1.5em',
    gridColumnStart: '1',
    gridColumnEnd: '3',
    gridRowStart: '2',
    gridRowEnd: '3',
    margin: '2em 0 0 0',
    background: colors.LIGHTEST_GRAY,
  },
  noStories: {
    padding: '1em',
    textAlign: 'center',
  },
})

export function Stories({ data: stories, location }) {
  if (stories) {
    if (stories.length === 0) {
      return <div className={styles.noStories}>No further items to display.</div>
    }

    const startingNumber = calculateStartingNumber(location.pathname)

    return (
      <div className={styles.storiesList} start={startingNumber}>
        {stories.filter(Boolean).map((story, index) => {
          const count = index === 0 ? startingNumber : index + startingNumber
          return (
            <article className={styles.storiesListItem} key={story.id}>
              <div className={styles.storiesListIndex}>{count}</div>
              <h2 className={styles.storiesListTitle}>
                <a href={story.url}>{story.title}</a>
              </h2>
              <div className={styles.storiesListByLine} suppressHydrationWarning>
                {story.score} points by {story.by.id}
              </div>
            </article>
          )
        })}
      </div>
    )
  }

  return (
    <div className={styles.storiesList}>
      {Array(20)
        .fill({})
        .map((_, index) => (
          <article className={classes(styles.storiesListItem, loadingAnimation)} key={index}>
            <div className={classes(styles.storiesListIndex, placeholder)}>{index + 1}</div>
            <div className={classes(styles.storiesListTitle, styles.storiesListTitlePlaceholder)}>
              <p className={placeholder}>placeholder title</p>
            </div>
            <div className={styles.storiesListByLine}>
              <p className={placeholder}>n points by placeholder</p>
            </div>
          </article>
        ))}
    </div>
  )
}
