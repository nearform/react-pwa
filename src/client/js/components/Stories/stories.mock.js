export const storiesMock = [
  {
    id: '17254222',
    title: 'Ions: Develop cloud applications by deploying to a running Datomic cluster',
    url: 'https://docs.datomic.com/cloud/ions/ions.html',
    score: 41,
    by: { id: 'simonpure' },
  },
  {
    id: '17254370',
    title: 'Einstein in Oxford',
    url: 'http://rsnr.royalsocietypublishing.org/content/early/2018/05/09/rsnr.2018.0002',
    score: 17,
    by: { id: 'sohkamyung' },
  },
  {
    id: '17252585',
    title: 'Cutter â€“ A Qt and C++ GUI for Radare2 reverse engineering framework',
    url: 'https://github.com/radareorg/cutter',
    score: 223,
    by: { id: 'xvilka' },
  },
  {
    id: '17254055',
    title: 'Caffeine dosing strategies to optimize alertness during sleep loss',
    url: 'https://onlinelibrary.wiley.com/doi/epdf/10.1111/jsr.12711',
    score: 31,
    by: { id: 'nickcw' },
  },
  {
    id: '17250697',
    title: 'Extracting the private key from a TREZOR with an oscilloscope (2015)',
    url: 'https://jochen-hoenicke.de/trezor-power-analysis/',
    score: 334,
    by: { id: 'mlejva' },
  },
]
