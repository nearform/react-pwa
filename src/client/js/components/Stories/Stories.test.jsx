import React from 'react'
import { render } from 'enzyme'

import { Stories } from './Stories'

import { storiesMock } from './stories.mock'

it('renders correctly when no data provided', () => {
  const props = {
    data: null,
    location: {
      pathname: '/',
    },
  }
  const test = render(<Stories {...props} />)
  expect(test).toMatchSnapshot()
})

it('renders correctly when empty data provided', () => {
  const props = {
    data: [],
    location: {
      pathname: '/',
    },
  }
  const test = render(<Stories {...props} />)
  expect(test).toMatchSnapshot()
})

it('renders correctly when data provided', () => {
  const props = {
    data: storiesMock,
    location: {
      pathname: '/',
    },
  }
  const test = render(<Stories {...props} />)
  expect(test).toMatchSnapshot()
})

it('renders correctly when data and page provided', () => {
  const props = {
    data: storiesMock,
    location: {
      pathname: '/foo/page/2',
    },
  }
  const test = render(<Stories {...props} />)
  expect(test).toMatchSnapshot()
})
