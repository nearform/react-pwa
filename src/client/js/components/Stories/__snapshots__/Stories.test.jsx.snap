// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when data and page provided 1`] = `
<ol
  class="storiesList_flhhfk0"
  start="31"
>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      31
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://docs.datomic.com/cloud/ions/ions.html"
      >
        Ions: Develop cloud applications by deploying to a running Datomic cluster
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      41 points by simonpure
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      32
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="http://rsnr.royalsocietypublishing.org/content/early/2018/05/09/rsnr.2018.0002"
      >
        Einstein in Oxford
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      17 points by sohkamyung
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      33
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://github.com/radareorg/cutter"
      >
        Cutter – A Qt and C++ GUI for Radare2 reverse engineering framework
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      223 points by xvilka
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      34
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://onlinelibrary.wiley.com/doi/epdf/10.1111/jsr.12711"
      >
        Caffeine dosing strategies to optimize alertness during sleep loss
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      31 points by nickcw
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      35
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://jochen-hoenicke.de/trezor-power-analysis/"
      >
        Extracting the private key from a TREZOR with an oscilloscope (2015)
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      334 points by mlejva
    </div>
  </li>
</ol>
`;

exports[`renders correctly when data provided 1`] = `
<ol
  class="storiesList_flhhfk0"
  start="1"
>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      1
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://docs.datomic.com/cloud/ions/ions.html"
      >
        Ions: Develop cloud applications by deploying to a running Datomic cluster
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      41 points by simonpure
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      2
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="http://rsnr.royalsocietypublishing.org/content/early/2018/05/09/rsnr.2018.0002"
      >
        Einstein in Oxford
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      17 points by sohkamyung
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      3
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://github.com/radareorg/cutter"
      >
        Cutter – A Qt and C++ GUI for Radare2 reverse engineering framework
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      223 points by xvilka
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      4
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://onlinelibrary.wiley.com/doi/epdf/10.1111/jsr.12711"
      >
        Caffeine dosing strategies to optimize alertness during sleep loss
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      31 points by nickcw
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli"
  >
    <div
      class="storiesListIndex_f1rppnmc"
    >
      5
    </div>
    <div
      class="storiesListTitle_fhylo1y"
    >
      <a
        href="https://jochen-hoenicke.de/trezor-power-analysis/"
      >
        Extracting the private key from a TREZOR with an oscilloscope (2015)
      </a>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      334 points by mlejva
    </div>
  </li>
</ol>
`;

exports[`renders correctly when empty data provided 1`] = `
<div
  class="noStories_f66kosg"
>
  No further items to display.
</div>
`;

exports[`renders correctly when no data provided 1`] = `
<ol
  class="storiesList_flhhfk0"
>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      1
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      2
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      3
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      4
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      5
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      6
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      7
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      8
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      9
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      10
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      11
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      12
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      13
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      14
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      15
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      16
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      17
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      18
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      19
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
  <li
    class="storiesListItem_f1h5zrli loading_fmjor1"
  >
    <div
      class="storiesListIndex_f1rppnmc placeholder_fu1ixcy"
    >
      20
    </div>
    <div
      class="storiesListTitle_fhylo1y storiesListTitlePlaceholder_fhbamv0"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        placeholder title
      </p>
    </div>
    <div
      class="storiesListByLine_f40jj2y"
    >
      <p
        class="placeholder_fu1ixcy"
      >
        n points by placeholder
      </p>
    </div>
  </li>
</ol>
`;
